respa_admin/views/reservations.py

```diff
@@ -0,0 +1,105 @@
+from django.contrib import messages
+from django.db.models import FieldDoesNotExist, Q
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import ListView
+
+from resources.models import Reservation, Resource
+from respa_admin.views.base import ExtraContextMixin
+
+
+class InvoiceableReservationListView(ExtraContextMixin, ListView):
+    model = Reservation
+    paginate_by = 10
+    context_object_name = "reservations"
+    template_name = "respa_admin/page_invoiceable_reservations.html"
+
+    def get(self, request, *args, **kwargs):
+        get_params = request.GET
+        self.search_query = get_params.get("search_query")
+        self.order_by = get_params.get("order_by", "name")
+        return super().get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context["search_query"] = self.search_query or ""
```

Comment:
You can also access these as {{ view.search_query }} etc so no need for the overridden method.

Reply:
I tried this but couldn't make it work right away for some reason. I'd leave this as is instead of spending time changing it. Good point, though.